File: app/src/main/java/com/song/deviceinfo/ui/base/NormalAdapter.java
package com.song.deviceinfo.ui.base;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.song.deviceinfo.R;

import androidx.annotation.NonNull;
import androidx.core.util.Pair;
import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by chensongsong on 2020/5/27.
 */
public class NormalAdapter extends BaseAdapter<Pair<String, String>, NormalAdapter.NormalHolder> {

    public NormalAdapter(Context context) {
        super(context);
    }

    @NonNull
    @Override
    public NormalHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View root = LayoutInflater.from(context).inflate(R.layout.item_normal, parent, false);
        return new NormalHolder(root);
    }

    @Override
    public void onBindViewHolder(@NonNull NormalHolder holder, int position) {
        if (position % 2 == 0) {
            holder.root.setBackgroundColor(context.getResources().getColor(R.color.color_white));
        } else {
            holder.root.setBackgroundColor(context.getResources().getColor(R.color.color_E8E8E8));
        }
        Pair<String, String> pair = data.get(position);
        String key = pair.first;
        String value = pair.second;
        holder.keyTv.setText(key);
        holder.valueTv.setText(value);
    }

    static class NormalHolder extends RecyclerView.ViewHolder {

        View root;
        TextView keyTv;
        TextView valueTv;

        public NormalHolder(View view) {
            super(view);
            root = view;
            keyTv = view.findViewById(R.id.tv_key);
            valueTv = view.findViewById(R.id.tv_value);
        }
    }
}

File: app/src/main/java/com/song/deviceinfo/ui/base/NormalAdapter.kt
package com.song.deviceinfo.ui.base

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.util.Pair
import androidx.recyclerview.widget.RecyclerView
import com.song.deviceinfo.R
import com.song.deviceinfo.ui.base.NormalAdapter.NormalHolder

/**
 * Created by chensongsong on 2020/5/27.
 */
class NormalAdapter(context: Context) : BaseAdapter<Pair<String, String>, NormalHolder>(context) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NormalHolder {
        val root = LayoutInflater.from(context).inflate(R.layout.item_normal, parent, false)
        return NormalHolder(root)
    }
    
    override fun onBindViewHolder(holder: NormalHolder, position: Int) {
        if (position % 2 == 0) {
            holder.root.setBackgroundColor(context.getColor(R.color.color_white))
        } else {
            holder.root.setBackgroundColor(context.getColor(R.color.color_E8E8E8))
        }
        val pair = items[position]
        holder.keyTv.text = pair.first
        holder.valueTv.text = pair.second
    }
    
    class NormalHolder(var root: View) : RecyclerView.ViewHolder(root) {
        var keyTv: TextView = root.findViewById(R.id.tv_key)
        var valueTv: TextView = root.findViewById(R.id.tv_value)
    }
}
